Source/Scene/HereMapsImageryProvider.js

```diff
@@ -0,0 +1,521 @@
+
+        var metadataError;
+        function metadataSuccess(data) {
+            try {
+                var jsonString = data;
+                var resource = JSON.parse(jsonString);
+
+                var attributionList = that._attributionList = resource.satellite;
+                if (!attributionList) {
+                    attributionList = that._attributionList = [];
+                }
+
+                for (var attributionIndex = 0, attributionLength = attributionList.length; attributionIndex < attributionLength; ++attributionIndex) {
+                    var attribution = attributionList[attributionIndex];
+
+                    attribution.credit = new Credit({
+                        text: attribution.alt
+                    });
+
+                    if (attribution.boxes) {
+                        var coverageAreas = attribution.boxes;
+
+                        for (var areaIndex = 0, areaLength = attribution.boxes.length; areaIndex < areaLength; ++areaIndex) {
+                            var area = coverageAreas[areaIndex];
+                            area.bbox = new Rectangle(
+                                    CesiumMath.toRadians(area[1]),
+                                    CesiumMath.toRadians(area[0]),
+                                    CesiumMath.toRadians(area[3]),
+                                    CesiumMath.toRadians(area[2]));
+                        }
+                    }
+                }
+
+                that._ready = true;
+                that._readyPromise.resolve(true);
+                TileProviderError.handleSuccess(metadataError);
+            }
+            catch (e)
+            {
+                metadataFailure(data);
+            }
+        }
+
+        function metadataFailure(data) {
+            var message = 'An error occurred while accessing ' + copyrightResource.url + '.';
+            metadataError = TileProviderError.handleError(metadataError, that, that._errorEvent, message, undefined, undefined, undefined, requestMetadata);
+            that._readyPromise.reject(new RuntimeError(message));
+        }
+
+        function requestMetadata() {
+            var metadata = copyrightResource.fetchJsonp('callback_func');
+            when(metadata, metadataSuccess, metadataFailure);
+        }
+
+        requestMetadata();
+    };
+
+    defineProperties(HereMapsImageryProvider.prototype, {
+        /**
+         * Gets the name of the HereMaps server url hosting the imagery.
+         * @memberof HereMapsImageryProvider.prototype
+         * @type {String}
+         */
+        url : {
+            get : function() {
+                return this._baseUrl;
+            }
+        },
+
+        /**
+         * Gets the proxy used by this provider.
+         * @memberof HereMapsImageryProvider.prototype
+         * @type {Proxy}
+         */
+        proxy : {
+            get : function() {
+                return this._proxy;
```

Comment:
This isn't set anywhere. It is actually a property of the resource. You can remove this property. Other providers have it for backwards compatibility reasons from pre-Resource days and will probably be deprecated.

Reply:
This gets set on line 86. But I assume this is no longer part of the ImageryProvider interface? I'll remove it.

Reply:
Oops. Forgot about the interface. That's why we haven't deprecated it yet. Keep it in there, but pass it into the resource as an option to `createIfNeeded`.

Reply:
OK I'll revert that change. It is passed in to the Resource on line 105. That resource is used to create all other requests. There is also a test case that covers usage of the proxy, and it's passing.

Reply:
I see that. If you look at Bing and the other providers, they no longer have a `_proxy` member. They do a `return this._resource.proxy;`